### ISSUE TITLE
Improving row statistics

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#4141]
Prepare > Column: Calculate > Row Summary calculates row statistics, i.e. it produces a new column with (say) the mean of existing columns.

1. It should have the option on what to do with missing values.  Currently if any values are missing then the summary is missing.
2. Like the ordinary calculations we should check what it does when a filter is applied.
3. There is a disabled button to define a user function.  Should that be added?
4. Can we consider weighted row summaries?

**Comment by shadrackkibet** [comment]
> It should have the option on what to do with missing values. Currently if any values are missing then the summary is missing.

@rdstern 
I see we can add ` na.rm` to the apply function. Should we have a checkbox somewhere  with the label **Ignore Missing Values**  by default unchecked?

**Comment by rdstern** [comment]
That's right.  We have that option on some other dialogues, e.g. Prepare > Column: Reshape > Summarise.  It could usefully be consistent with that.

**Comment by dannyparsons** [comment]
Yes I needed this the other day!

**Comment by rdstern** [comment]
@shadrackkibet can this be closed?

**Comment by shadrackkibet** [comment]
**2,3 and 4** are not yet sorted out. So, this cannot be closed.

**Comment by dannyparsons** [comment]
I think 1 was the most urgent so this can move to PENDING.

**Comment by shadrackkibet** [comment]
> It should have the option on what to do with missing values. Currently if any values are missing then the summary is missing.

I think we can make the missing option more robust here by allowing the user to define missing value options like what we do in climatic summaries and the other ordinary summaries dialog. I needed this facility for some staff working at KMD who are using this dialog. I think it is now time to enable this feature.

**Comment by lilyclements** [comment]
@rdstern:

> Like the ordinary calculations we should check what it does when a filter is applied.

This currently throws an error. To prevent this can we temporarily add in `use_current_filter=FALSE` to the `data_book$get_data_frame` argument. This is what we do in the calculator dialog at the moment.


> Can we consider weighted row summaries?

We should have this, but I'm not sure how we can do it. Could we have another option on this where you put in the weighting for the columns given?

**Comment by lilyclements** [comment]
@rdstern great. I've done (a) and (b). We can work on this later on with a filter.
